front/src/applications/stdcm/utils/consistValidation.ts

```diff
@@ -0,0 +1,88 @@
+import { kgToT } from 'utils/physics';
+
+export const CONSIST_TOTAL_MASS_MAX = 10000; // ton
+export const CONSIST_TOTAL_LENGTH_MAX = 750; // m
+export const CONSIST_MAX_SPEED_MIN = 30; // km/h
+
+export const validateTotalMass = ({
+  tractionEngineMass = 0,
+  towedMass = 0,
+  totalMass,
+}: {
+  tractionEngineMass?: number;
+  towedMass?: number;
+  totalMass?: number;
+}) => {
+  if (!totalMass) {
+    return undefined;
+  }
+
+  if (totalMass <= 0) {
+    return 'consist.errors.totalMass.negative';
+  }
+
+  const tractionMassInTons = kgToT(tractionEngineMass);
+  const consistMassInTons = kgToT(tractionEngineMass + towedMass);
+  const massLimit = towedMass ? consistMassInTons : tractionMassInTons;
+
+  if (totalMass < massLimit || totalMass >= CONSIST_TOTAL_MASS_MAX) {
+    return 'consist.errors.totalMass.range';
+  }
+
+  if (
+    totalMass < (towedMass ? consistMassInTons : tractionMassInTons) ||
+    totalMass >= CONSIST_TOTAL_MASS_MAX
+  ) {
+    return 'consist.errors.totalMass.range';
+  }
```

Comment:
Sounds like this is a duplicate check? The `if` right above seems to check for the same thing.

Reply:
It check the lower and higher bounds